packages/alpinejs/src/directives/x-model.js

```diff
@@ -59,22 +59,45 @@ directive('model', (el, { modifiers, expression }, { effect, cleanup }) => {
     // If the element we are binding to is a select, a radio, or checkbox
     // we'll listen for the change event instead of the "input" event.
     var event = (el.tagName.toLowerCase() === 'select')
-        || ['checkbox', 'radio'].includes(el.type)
-        || modifiers.includes('lazy')
-            ? 'change' : 'input'
+    || ['checkbox', 'radio'].includes(el.type)
+    || modifiers.includes('lazy')
+        ? 'change' : 'input'
 
     // We only want to register the event listener when we're not cloning, since the
     // mutation observer handles initializing the x-model directive already when
     // the element is inserted into the DOM. Otherwise we register it twice.
     let removeListener = isCloning ? () => {} : on(el, event, modifiers, (e) => {
         setValue(getInputValue(el, modifiers, e, getValue()))
     })
-    
-    if (modifiers.includes('fill'))
-        if ([null, ''].includes(getValue())
-            || (el.type === 'checkbox' && Array.isArray(getValue()))) {
+
+    if (modifiers.includes('fill')) {
+        // autofill x-data object
+        if (typeof expression === 'string') {
+            let parent = el.closest('[x-data]');
+            if (parent) {
+                let xData = Alpine.$data(parent);
+                if (typeof xData[expression] === 'undefined') {
```

Comment:
This will incorrectly trigger this branch if the model is an expression and not a literal key (ie. 'info.query')